File: c++/basics/io.cpp
// Soma durante a leitura: não precisa guardar os valores.
// Entrada: N e N inteiros; exemplo: 4 1 2 3 4 -> saída: 10.
// Tempo O(N), memória O(1). N >= 0; soma e valores devem caber em long long.
#include <iostream>
int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);
    int n;
    if (!(std::cin >> n)) return 0;
    long long total = 0;
    for (int i = 0; i < n; ++i) {
        long long x; std::cin >> x;
        total += x;
    }
    std::cout << total << '\n';
}

File: python/basics/io.py
"""Entrada por tokens quando as quebras de linha não importam.
Use para somar N valores, mesmo distribuídos em várias linhas.
Entrada: 4\n1 2\n3 4\n. Saída: 10.
Tempo O(N), memória O(N) para a leitura completa; não usar em interativos.
Para economizar memória, processe linhas com sys.stdin.readline.
"""
import sys


def main():
    tokens = iter(sys.stdin.buffer.read().split())
    first = next(tokens, None)
    if first is None:
        return
    n = int(first)
    total = sum(int(next(tokens)) for _ in range(n))
    print(total)


if __name__ == "__main__":
    main()
